src/syrupy/extensions/amber/serializer.py

```diff
@@ -87,43 +90,60 @@ def write_file(
             for snapshot in sorted(snapshot_collection, key=lambda s: s.name):
                 snapshot_data = str(snapshot.data)
                 if snapshot_data is not None:
-                    f.write(f"{cls._marker_name} {snapshot.name}\n")
+                    f.write(f"{cls.Marker.Comment}{cls.Marker.Name}: {snapshot.name}\n")
                     for data_line in snapshot_data.splitlines(keepends=True):
                         f.write(cls.with_indent(data_line, 1))
-                    f.write(f"\n{cls._marker_divider}\n")
+                    f.write(f"\n{cls.Marker.Comment}{cls.Marker.Divider}\n")
 
     @classmethod
-    def read_file(cls, filepath: str) -> "SnapshotCollection":
-        """
-        Read the raw snapshot data (str) from the snapshot file into a dict
-        of snapshot name to raw data. This does not attempt any deserialization
-        of the snapshot data.
-        """
-        name_marker_len = len(cls._marker_name)
+    def __read_file_with_markers(
+        cls, filepath: str
+    ) -> Generator["Snapshot", None, None]:
+        marker_offset = len(cls.Marker.Comment)
         indent_len = len(cls._indent)
-        snapshot_collection = SnapshotCollection(location=filepath)
+
+        test_name = None
+        snapshot_data = ""
+
         try:
             with open(filepath, "r", encoding=TEXT_ENCODING, newline=None) as f:
-                test_name = None
-                snapshot_data = ""
                 for line in f:
-                    if line.startswith(cls._marker_name):
-                        test_name = line[name_marker_len:].strip(f" {cls._marker_crn}")
-                        snapshot_data = ""
-                        continue
-                    elif test_name is not None:
-                        if line.startswith(cls._indent):
-                            snapshot_data += line[indent_len:]
-                        elif line.startswith(cls._marker_divider) and snapshot_data:
-                            snapshot_collection.add(
-                                Snapshot(
+                    if line.startswith(cls.Marker.Comment):
+                        marker_key, *marker_rest = line[marker_offset:].split(
+                            ":", maxsplit=1
+                        )
+                        marker_key = marker_key.rstrip(" \r\n")
+                        marker_value = marker_rest[0] if marker_rest else None
+
+                        if marker_key == cls.Marker.Name:
```

Comment:
Makes it easier to reason about arbitrary markers. Planning on introducing a new marker for index shortly